extensions/git/src/git.ts

```diff
@@ -1979,6 +1979,15 @@ export class Repository {
 
 	async getHEAD(): Promise<Ref> {
 		try {
+			// Attempt for parse the HEAD file
+			const refsFromHEADFileRaw = await fs.readFile(path.join(this.dotGit.commonPath ?? this.dotGit.path, 'HEAD'), 'utf8');
+			const refsFromHEADFile = refsFromHEADFileRaw.split(/\r?\n/g, 1);
+
+			if (refsFromHEADFile[0].startsWith('ref: refs/heads/')) {
+				return { name: refsFromHEADFile[0].substring(16).trim(), commit: undefined, type: RefType.Head };
+			}
```

Comment:
Right now, this is splitting strings, doing string comparisons and more splitting and trimming. Could we perhaps have a single regex which, in a single `exec()` call, would both validate and extract the head name?

Reply:
Agreed. Fixed.